src/rotation.zig

```diff
@@ -321,6 +321,173 @@ pub const RotationModes = struct {
 			return true;
 		}
 	};
+	pub const Branch = struct { // MARK: Branch
+		pub const id: []const u8 = "branch";
+		pub const dependsOnNeighbors = true;
+		var branchModels: std.StringHashMap(u16) = undefined;
+		const BranchData = packed struct(u6) {
+			enabledConnections: u6,
+
+			pub fn init(blockData: u16) BranchData {
+				return .{.enabledConnections = @truncate(blockData)};
+			}
+
+			pub fn isConnected(self: @This(), neighbor: Neighbor) bool {
+				return (self.enabledConnections & Neighbor.bitMask(neighbor)) != 0;
+			}
+
+			pub fn setConnection(self: *@This(), neighbor: Neighbor, value: bool) void {
+				if(value) {
+					self.enabledConnections |= Neighbor.bitMask(neighbor);
+				} else {
+					self.enabledConnections &= ~Neighbor.bitMask(neighbor);
+				}
+			}
+		};
+
+		fn init() void {
+			branchModels = .init(main.globalAllocator.allocator);
+		}
+
+		fn deinit() void {
+			branchModels.deinit();
+		}
+
+		fn branchTransform(quad: *main.models.QuadInfo, data: BranchData) void {
+			for(&quad.corners) |*corner| {
+				// Ain't no way we are making snake out of this one.
+				// zig fmt: off
+				if(
+					(!data.isConnected(Neighbor.dirNegX) and corner[0] == 0)
+					or (!data.isConnected(Neighbor.dirPosX) and corner[0] == 1)
+					or (!data.isConnected(Neighbor.dirNegY) and corner[1] == 0)
+					or (!data.isConnected(Neighbor.dirPosY) and corner[1] == 1)
+					or (!data.isConnected(Neighbor.dirDown) and corner[2] == 0)
+					or (!data.isConnected(Neighbor.dirUp) and corner[2] == 1)
+				) return degenerateQuad(quad);
+				// zig fmt: on
+			}
+		}
+
+		fn degenerateQuad(quad: *main.models.QuadInfo) void {
+			for(&quad.corners) |*corner| {
+				corner[0] = 0.5;
+				corner[1] = 0.5;
+				corner[2] = 0.5;
```

Comment:
Just in case you were not aware of it:
You could also use `corner.* = @splat(0.5)` to fill the entire thing in one statement.